Persephone/Controllers/WindowController.swift

```diff
@@ -26,28 +28,108 @@ class WindowController: NSWindowController {
       name: Notification.stateChanged,
       object: AppDelegate.mpdClient
     )
+
+    NotificationCenter.default.addObserver(
+      self,
+      selector: #selector(timeChanged(_:)),
+      name: Notification.timeChanged,
+      object: AppDelegate.mpdClient
+    )
+
+    trackProgress.font = .timerFont
+    trackRemaining.font = .timerFont
   }
 
   @objc func stateChanged(_ notification: Notification) {
     guard let state = notification.userInfo?[Notification.stateKey] as? MPDClient.Status.State
       else { return }
 
-    setTransportControlState(state)
+    self.state = state
+
+    setTransportControlState()
   }
 
-  func setTransportControlState(_ state: MPDClient.Status.State) {
+  func setTransportControlState() {
+    guard let state = state else { return }
+
     transportControls.setEnabled(state.isOneOf([.playing, .paused]), forSegment: 0)
     transportControls.setEnabled(state.isOneOf([.playing, .paused, .stopped]), forSegment: 1)
     transportControls.setEnabled(state.isOneOf([.playing, .paused]), forSegment: 2)
     transportControls.setEnabled(state.isOneOf([.playing, .paused]), forSegment: 3)
 
     if state.isOneOf([.paused, .stopped, .unknown]) {
-      transportControls.setImage(playIcon, forSegment: 1)
+      transportControls.setImage(.playIcon, forSegment: 1)
+    } else {
+      transportControls.setImage(.pauseIcon, forSegment: 1)
+    }
+  }
+
+  @objc func timeChanged(_ notification: Notification) {
+    guard let totalTime = notification.userInfo?[Notification.totalTimeKey] as? UInt,
+      let elapsedTimeMs = notification.userInfo?[Notification.elapsedTimeMsKey] as? UInt
+      else { return }
+
+    self.totalTime = totalTime
+    self.elapsedTimeMs = elapsedTimeMs
+
+    setTrackProgressControls()
+  }
+
+  func setTrackProgressControls() {
```

Comment:
This method gets called when the server returns a status update. This doesn't happen every time the elapsed time changes so we are expected to keep time by ourselves until we get another update.

Interestingly I've noticed that this can get out of sync with the time on the server when the app is backgrounded, presumably because it starts throttling the timer. Not sure on the best way to deal with this yet.